### PR TITLE
Adding GoAlert Notification

### PR DESCRIPTION
**Pull request by mhkarimi1383**
# Description

Adding [GoAlert](https://goalert.me/) support

according to [Alerting From Generic API](https://goalert.me/docs/alerting/alerting-from-generic-api/) and it's handler [here](https://github.com/target/goalert/blob/master/genericapi/handler.go)

## Type of change

Please delete any options that are not relevant.

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update (if other notifications are documented) #284

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

Screenshot from 2022-08-24 00-03-08

### CODE REVIEW

**Comment by mhkarimi1383** [comment]
Everything Should be fine my reviews and tests will be finished tomorrow :)

**Comment by mhkarimi1383** [comment]
Hello again
`json` content type was not working with GoAlert correctly
So, I migrated to `form-data`
but `form-data` and `axios` where cannot work great together but new version of `axios` has internal support for `form-data` handling, so I upgraded `axios` package

more info:
[axios form-data](https://www.npmjs.com/package/axios#form-data)

File: server/notification-providers/goalert.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                data["action"] = closeAction;
+            }
+            let headers = {
+                "Content-Type": "multipart/form-data",
+            };
+            let config = {
+                headers: headers
+            };
+            let resp = await axios.post(`${notification.goAlertBaseURL}/api/v2/generic/incoming?token=${notification.goAlertToken}`, data, config);
+            console.log(resp);
```
Comment (Saibamen) [inline server/notification-providers/goalert.js:26]:
```suggestion
```

File: src/components/notifications/index.js [hunk 2]
```diff
@@ -81,6 +82,7 @@ const NotificationFormList = {
     "telegram": Telegram,
     "webhook": Webhook,
     "WeCom": WeCom,
+    "GoAlert": GoAlert
```
Comment (Saibamen) [inline src/components/notifications/index.js:85]:
```suggestion
    "GoAlert": GoAlert,
```

**Review by Saibamen — CHANGES_REQUESTED** [review]

**Review by Saibamen — APPROVED** [review]

**Comment by chakflying** [comment]
You should be running npm `v7+` to use the newer version of the lockfile. The node version should not matter as long as it's `14+`.

**Comment by mhkarimi1383** [comment]
using npm `8.18.0`
but ci fails

**Comment by k3rnelpan1c-dev** [comment]
IIRC Node 14 + NPM 7 (the default NPM version with Node 14 IIRC), does not support lockfiles v2.
The general CI `auto-test` use a range of Node Version [14..18] but the `check-linters`only runs 14.

Side note: 14 being a prior LTS is still in maiintainac till May 2023 (https://nodejs.org/en/about/releases/)

**Comment by chakflying** [comment]
Thanks, I remembered it wrong.

Thing is, CI should still work even with lockfile version mismatch (as it has worked all this time).  I think you broke something else when you regenerated that file on MacOS, somehow the linux dependencies are missing. Anyway I think louislam can handle it if this gets merged.

**Comment by mhkarimi1383** [comment]
> Thanks, I remembered it wrong.
> 
> Thing is, CI should still work even with lockfile version mismatch (as it has worked all this time). I think you broke something else when you regenerated that file on MacOS, somehow the linux dependencies are missing. Anyway I think louislam can handle it if this gets merged.

I'm on ArchLinux using `nvm`

**Comment by mhkarimi1383** [comment]
```
node --version
v14.6.0
```

```
npm --version
6.14.6
```

What is the ideal version?

**Comment by k3rnelpan1c-dev** [comment]
okay, took another look, and seems as if the main branch is already using a v2 lock-file. so I am surprised that this hasn't pop up earlier :thinking:

**Comment by mhkarimi1383** [comment]
> okay, took another look, and seems as if the main branch is already using a v2 lock-file. so I am surprised that this hasn't pop up earlier :thinking: 
> 
> Edit:
> 
> :man_facepalming: we where looking at the wrong line (like most regular npm users and the lockfile was just a warning not an error):
> 
> > npm ERR! code EBADPLATFORM
> > npm ERR! notsup Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} > (current: {"os":"linux","arch":"x64"})
> > npm ERR! notsup Valid OS:    darwin
> > npm ERR! notsup Valid Arch:  any
> > npm ERR! notsup Actual OS:   linux
> > npm ERR! notsup Actual Arch: x64
> 
> looks like `fsevents@2.3.2` wants the host to be MacOS, yet we on Linux CI and in the container images too.

How main branch is working?

**Comment by k3rnelpan1c-dev** [comment]
I have not the slightest idea, but a couple of pointers that may be worth evaluating if they are the cause:

* flaky CI (given you are on Arch you could try nuking your current downloaded dependency cache and attempt build to see if that theory applies)
* some funky dependency change within a sub-dependency (if any of the dependencies uses version ranges or similar)

(given that CI passed before regenerating the lockfile it can be either)

Please note that I am merely a CI enthusiast who likes this project and stumbled over your PR. I have no power here, besides trying to help debugging :thinking:

**Comment by mhkarimi1383** [comment]
fixed by switching to node v16

**Pull request merged**